src/main/java/com/authservice/config/UserInfoUserDetails.java

```diff
@@ -0,0 +1,65 @@
+
+package com.authservice.config;
+import com.authservice.model.User;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Data
+@NoArgsConstructor
+public class UserInfoUserDetails implements UserDetails {
+
+    private String name;
+    private String password;
+    private List<GrantedAuthority> authorities;
+
+    public UserInfoUserDetails(User userInfo) {
+        name = userInfo.getName();
+        password = userInfo.getPassword();
+        authorities = Arrays.stream(userInfo.getRoles().split(","))
```

Comment:
this.name = userInfo.getName();
       this. password = userInfo.getPassword();
       this. authorities = Arrays.stream(userInfo.getRoles().split(","))

Reply:
updated code changes : completed